src/nvim/option.c

```diff
@@ -524,11 +524,15 @@ char *get_lib_dir(void)
 ///
 /// Windows: Uses "…/nvim-data" for kXDGDataHome to avoid storing
 /// configuration and data files in the same path. #4403
-static void set_runtimepath_default(void)
+///
+/// If "clean_arg" is true, Nvim was started with --clean.
+static void set_runtimepath_default(bool clean_arg)
 {
   size_t rtp_size = 0;
   char *const data_home = stdpaths_get_xdg_var(kXDGDataHome);
-  char *const config_home = stdpaths_get_xdg_var(kXDGConfigHome);
+  char *const config_home = clean_arg
+    ? NULL
+    : stdpaths_get_xdg_var(kXDGConfigHome);
```

Comment:
This removes `stdpath('config')` from `&runtimepath`. Not sure if `stdpath('data')` and other directories should be excluded as well.

Reply:
Yeah, all of the user-specific config should be ignored.  Only the system-wide config should be available.